sphinx/directives/code.py

```diff
@@ -53,8 +53,15 @@ def run(self):
         if 'linenothreshold' in self.options:
             try:
                 linenothreshold = int(self.options['linenothreshold'])
-            except Exception:
-                linenothreshold = 10
+            except ValueError:
+                default_linenothreshold = 10
+                location = self.state_machine.get_source_and_line(self.lineno)
+                logger.warning(__('linenothreshold value must be int typed. '
+                                  'Passed %r. Using default value (%d).'
```

Comment:
`10` is not default value. Please see "else" clause. `sys.maxsize` is default value.
I don't know why `10` is used for fallback.

Reply:
This number was appeared since the beginning of `linenothreshold` option (f2ff0419bd).

I think this number is meaningless. I think fallback to `sys.maxint` is better.